modules/http-api/src/main/java/org/mule/runtime/http/api/domain/entity/HttpEntity.java

```diff
@@ -6,11 +6,70 @@
  */
 package org.mule.runtime.http.api.domain.entity;
 
+import org.mule.runtime.http.api.domain.entity.multipart.HttpPart;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+
 /**
- * Marker interface for a HTTP message entity.
+ * An entity that can be sent or received via an {@link org.mule.runtime.http.api.domain.message.request.HttpRequest} or
+ * {@link org.mule.runtime.http.api.domain.message.response.HttpResponse}.
+ ** <p>
+ * There are three distinct types of entities, depending on their content:
+ * <ul>
+ * <li><b>streamed</b>: The content is received from a stream.
+ *     </li>
+ * <li><b>composed</b>: The content is made of several parts.
+ *     </li>
+ * <li><b>neither of the above</b>: The content is in memory and whole.
+ *     </li>
+ * </ul>
+ * These will determine how the content can be accessed although there may overlap.
  *
  * @since 4.0
  */
 public interface HttpEntity {
 
+  /**
+   * Tells whether this entity's content is stream based. Streamed entities can only provide their content once, regardless of the
+   * access method.
+   *
+   * @return {@code true} if content is streamed, {@code false} otherwise
+   * @see #getContent()
+   */
+  boolean isStreaming();
+
+  /**
+   * Tells whether or not this entity is composed of several parts, in which case they should be available through {@link #getParts()}.
+   *
+   * @return {@code true} if there are several content parts, {@code false} otherwise
+   */
+  boolean isComposed();
+
+  /**
+   * Provides the entity's content as a stream. All streaming entities should provide their stream.
+   *
+   * @return an {@code InputStream} representing this entity's content or {@code null} if such representation is not possible
+   */
+  InputStream getContent();
+
+  /**
+   * Provides the entity's content as bytes. If the entity is stream based, then the stream will be consumed as a consequence.
+   *
+   * @return a byte array representing this entity's content or {@code null} if such representation is not possible
```

Comment:
`or {@code null} if such representation is not possible` -> isn't the implementation returning an emty array in this case?